crates/router/src/connector/payme/transformers.rs

```diff
@@ -397,59 +454,82 @@ impl<F>
             types::PaymentsResponseData,
         >,
     ) -> Result<Self, Self::Error> {
-        let currency_code = item.data.request.get_currency()?;
-        let amount = item.data.request.get_amount()?;
-        let amount_in_base_unit = utils::to_currency_base_unit(amount, currency_code)?;
-        let pmd = item.data.request.payment_method_data.to_owned();
-        let payme_auth_type = PaymeAuthType::try_from(&item.data.connector_auth_type)?;
-
-        let session_token = match pmd {
-            Some(PaymentMethodData::Wallet(
-                api_models::payments::WalletData::ApplePayThirdPartySdk(_),
-            )) => Some(api_models::payments::SessionToken::ApplePay(Box::new(
-                api_models::payments::ApplepaySessionTokenResponse {
-                    session_token_data:
-                        api_models::payments::ApplePaySessionResponse::NoSessionResponse,
-                    payment_request_data: Some(api_models::payments::ApplePayPaymentRequest {
-                        country_code: item.data.get_billing_country()?,
-                        currency_code,
-                        total: api_models::payments::AmountInfo {
-                            label: "Apple Pay".to_string(),
-                            total_type: None,
-                            amount: amount_in_base_unit,
+        match item.data.auth_type {
```

Comment:
why do we need this match statement? irrespective of the auth_type we are calling generate-sale in the flow

Reply:
it is to differentiate between the 3ds and non 3ds flow, since we are breaking at preprocessing at 3ds, we need  redirection for 3ds. To show redirection we need authenticationpending status too. But in case of of non 3ds, we don't have redirection at preprocessing step, rather we move to authorize flow to complete the payment